realm/src/androidTest/java/io/realm/DynamicRealmListTest.java

```diff
@@ -0,0 +1,205 @@
+        obj.setFieldDouble(1.234d);
+        obj.setFieldBinary(new byte[]{1, 2, 3});
+        obj.setFieldBoolean(true);
+        obj.setFieldDate(new Date(1000));
+        obj.setFieldObject(obj);
+        obj.getFieldList().add(obj);
+        dynamicObject = new DynamicRealmObject(obj);
+        dynamicList = dynamicObject.getList(AllJavaTypes.FIELD_LIST);
+        realm.commitTransaction();
+    }
+
+    @Override
+    protected void tearDown() throws Exception {
+        super.tearDown();
+        realm.close();
+    }
+
+    public void testAddNullObjectThrows() {
+        realm.beginTransaction();
+        try {
+            dynamicList.add(null);
+            fail();
+        } catch (IllegalArgumentException ignored) {
+        } finally {
+            realm.cancelTransaction();
+        }
+    }
+
+    public void testAddWrongTableClassThrows() {
+        realm.beginTransaction();
+        Dog dog = realm.createObject(Dog.class);
+        try {
+            dynamicList.add(new DynamicRealmObject(dog));
+            fail();
+        } catch (IllegalArgumentException ignored) {
+        } finally {
+            realm.cancelTransaction();
+        }
+    }
+
+    public void testAddFromWrongRealmThrows() {
+        RealmConfiguration otherConfig = new RealmConfiguration.Builder(getContext()).name("realm2").build();
+        Realm.deleteRealm(otherConfig);
+        Realm realm2 = Realm.getInstance(otherConfig);
+        realm2.beginTransaction();
+        AllJavaTypes realm2Object = realm2.createObject(AllJavaTypes.class);
+        realm2.commitTransaction();
+
+        realm.beginTransaction();
+        try {
+            dynamicList.add(new DynamicRealmObject(realm2Object));
+            fail();
+        } catch (IllegalArgumentException ignored) {
+        } finally {
+            realm.cancelTransaction();
+            realm2.close();
+        }
+    }
+
+    public void testAddObject() {
+        realm.beginTransaction();
+        AllJavaTypes obj = realm.createObject(AllJavaTypes.class);
+        assertEquals(1, dynamicList.size());
+        dynamicList.add(new DynamicRealmObject(obj));
+        realm.commitTransaction();
+
+        assertEquals(2, dynamicList.size());
+    }
+
+    public void testClear() {
+        realm.beginTransaction();
+        dynamicList.clear();
+        realm.commitTransaction();
+
+        assertEquals(0, dynamicList.size());
+    }
+
+    public void testGetIllegalIndexThrows() {
+        int[] indexes = new int[] { -1, 1 };
+        for (int i : indexes) {
+            try {
+                dynamicList.get(i);
+                fail("Could retrieve from index" + i);
+            } catch (IndexOutOfBoundsException ignored) {
+            }
+        }
+    }
+
+    public void testGet() {
+        DynamicRealmObject listObject = dynamicList.get(0);
+        assertEquals(dynamicObject, listObject);
+    }
+
+    public void testSetIllegalLocationThrows() {
+        int[] indexes = new int[] { -1, 1 };
```

Comment:
Oh, yes, I want to argue that exactly. I still would like to have that to avoid confusion in the future  :smile: